pkg/apis/policy/v1alpha1/types_certificaterequestpolicy.go

```diff
@@ -219,6 +233,44 @@ type CertificateRequestPolicyAllowedString struct {
 	// Defaults to `false`.
 	// +optional
 	Required *bool `json:"required,omitempty"`
+
+	// Validations applies rules using Common Expression Language (CEL) to
+	// validate attribute value present on request beyond what is possible
+	// to express using value/required.
+	// An attribute value on the related CertificateRequest field must pass
+	// ALL validations for the request to be granted by this policy.
+	//
+	// NOTE: Validations will only be performed after a request is considered
+	// granted by this policy value/required fields. This behavior might
+	// change in a future version of the API.
+	// +listType=map
+	// +listMapKey=rule
+	// +optional
+	Validations []ValidationRule `json:"validations,omitempty"`
+}
+
+// ValidationRule describes a validation rule expressed in CEL.
+type ValidationRule struct {
+	// Rule represents the expression which will be evaluated by CEL.
+	// ref: https://github.com/google/cel-spec
+	// The Rule is scoped to the location of the validations in the schema.
+	// The `self` variable in the CEL expression is bound to the scoped value.
+	// To enable more advanced validation rules, approver-policy provides the
+	// `cr` (map) variable to the CEL expression containing `namespace` and
+	// `name` of the `CertificateRequest` resource.
+	//
+	// Examples:
+	// - Rule for namespaced DNSNames:
+	//	 {"rule": "self.endsWith(cr.namespace + '.svc.cluster.local'"}
+	Rule string `json:"rule"`
+
+	// Message represents the message displayed when validation fails.
```

Comment:
Is this an output or an input?

`represents` feels like an output, but I think it's an input.

Reply:
In the policy API it is an input, but it sets a message that will provide output to the user in case the rule fails. 😉

Reply:
```suggestion
	// Message is the message to display when validation fails.
```